src/samplereader/FragmentedSampleReader.cpp

```diff
@@ -408,16 +408,17 @@ AP4_Result CFragmentedSampleReader::ProcessMoof(AP4_ContainerAtom* moof,
       }
 
       // Inform decrypter of pattern decryption (CBCS)
-      if (m_protectedDesc->GetSchemeType() == AP4_PROTECTION_SCHEME_TYPE_CENC)
+      AP4_UI32 schemeType = m_protectedDesc->GetSchemeType();
+      if (schemeType == AP4_PROTECTION_SCHEME_TYPE_CENC || schemeType == AP4_PROTECTION_SCHEME_TYPE_CBCS)
       {
-        m_singleSampleDecryptor->SetEncryptionScheme(ENCRYPTION_SCHEME::CENC);
-        m_singleSampleDecryptor->SetCrypto(0, 0);
-      }
-      else if (m_protectedDesc->GetSchemeType() == AP4_PROTECTION_SCHEME_TYPE_CBCS)
-      {
-        m_singleSampleDecryptor->SetEncryptionScheme(ENCRYPTION_SCHEME::CBCS);
-        m_singleSampleDecryptor->SetCrypto(sample_table->GetCryptByteBlock(),
-                                           sample_table->GetSkipByteBlock());
+        m_readerCryptoInfo.m_cryptBlocks = sample_table->GetCryptByteBlock();
+        m_readerCryptoInfo.m_skipBlocks = sample_table->GetSkipByteBlock();
+        m_readerCryptoInfo.m_mode = schemeType == AP4_PROTECTION_SCHEME_TYPE_CENC
+                                        ? AP4_CENC_CIPHER_AES_128_CTR
+                                        : AP4_CENC_CIPHER_AES_128_CBC;
```

Comment:
expand it its better readable
```suggestion
if (schemeType == AP4_PROTECTION_SCHEME_TYPE_CENC)
        m_readerCryptoInfo.m_mode = AP4_CENC_CIPHER_AES_128_CTR;
else
       m_readerCryptoInfo.m_mode = AP4_CENC_CIPHER_AES_128_CBC;
```